powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/v3/OperationServiceBehavior.java

```diff
@@ -575,10 +577,21 @@ private boolean isDataEqual(@NotNull OperationEntity operation, String providedD
     @Transactional
     public void expireOperations() {
         LockAssert.assertLocked();
-        final Date currentTimestamp = new Date();
         logger.debug("Running scheduled task for expiring operations");
-        try (final Stream<OperationEntity> pendingOperations = operationRepository.findExpiredPendingOperations(currentTimestamp)) {
-            pendingOperations.forEach(op -> expireOperation(op, currentTimestamp));
-        }
+
+        final Date currentTimestamp = new Date();
+        List<String> expiringOperationIds = operationRepository.findExpiredPendingOperationIds(currentTimestamp);
+        if (expiringOperationIds.isEmpty())
+            return;
+        int updatedRows = operationRepository.setExpiredToPendingOperations(expiringOperationIds);
+        if (expiringOperationIds.size() > updatedRows)
+            logger.warn("There was less updated rows ({}) than was expected ({}), ids on input: {}", expiringOperationIds.size(), updatedRows, expiringOperationIds);
```

Comment:
We do not need to log this message, it can easily happen that the number of updated records is different, because data in the database can change. If you want to keep the message, let's switch to `debug` level.